{
  "openapi": "3.0.0",
  "servers": [],
  "info": {
    "version": "3.2",
    "title": "ExecService",
    "description": "Azure Dev Spaces ExecService v3.2"
  },
  "paths": {
    "/api/v3.2/AzdsTraces": {
      "post": {
        "tags": [
          "AzdsTraces"
        ],
        "operationId": "AzdsTraces_PostAzdsUserClusterLogsAsync",
        "parameters": [
          {
            "name": "onbehalfof",
            "in": "query",
            "required": false,
            "schema": {
              "type": "string",
              "default": "UserClusterLog"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "The request was successful"
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure. The request failed. Bad request.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Build/spaces/{spaceName}/services/{serviceName}/log": {
      "post": {
        "tags": [
          "Build"
        ],
        "operationId": "Build_PostBuildLogs",
        "parameters": [
          {
            "name": "spaceName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          },
          {
            "name": "serviceName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          },
          {
            "name": "requestId",
            "in": "query",
            "required": false,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Success"
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure. The request failed. Bad request.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        },
        "requestBody": {
          "required": false,
          "x-ms-requestBody-name": "log",
          "content": {
            "application/json-patch+json": {
              "schema": {
                "type": "string"
              }
            },
            "application/json": {
              "schema": {
                "type": "string"
              }
            },
            "text/json": {
              "schema": {
                "type": "string"
              }
            },
            "application/*+json": {
              "schema": {
                "type": "string"
              }
            }
          }
        },
        "x-ms-requestBody-index": 3
      }
    },
    "/api/v3.2/Config/{path}": {
      "get": {
        "tags": [
          "Config"
        ],
        "operationId": "Config_GetConfigDocument",
        "parameters": [
          {
            "name": "path",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "The request was successful.",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "string"
                }
              },
              "application/json": {
                "schema": {
                  "type": "string"
                }
              },
              "text/json": {
                "schema": {
                  "type": "string"
                }
              }
            }
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure. The request failed. Bad request.",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Liveness": {
      "get": {
        "tags": [
          "Liveness"
        ],
        "operationId": "Liveness_LivenessCheck",
        "responses": {
          "200": {
            "description": "The liveness check was successful"
          },
          "default": {
            "description": "Error response describing the reason for operation failure. The service has run out of storage space",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Ping/auth": {
      "get": {
        "tags": [
          "Ping"
        ],
        "operationId": "Ping_CheckUserAuth",
        "responses": {
          "200": {
            "description": "Success"
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Ping/auth/whitelist": {
      "get": {
        "tags": [
          "Ping"
        ],
        "operationId": "Ping_CheckWhitelistAndUserAuth",
        "responses": {
          "200": {
            "description": "Success"
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Ping/auth/whitelistonly": {
      "get": {
        "tags": [
          "Ping"
        ],
        "operationId": "Ping_CheckWhitelistOnlyAuth",
        "responses": {
          "200": {
            "description": "Success"
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Ping": {
      "get": {
        "tags": [
          "Ping"
        ],
        "operationId": "Ping_PingCheck",
        "responses": {
          "200": {
            "description": "The ping was successful",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/PingResponse"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/PingResponse"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/PingResponse"
                }
              }
            }
          },
          "default": {
            "description": "Error response describing the reason for operation failure. The service cannot connect to one of its dependencies",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Spaces/{spaceName}": {
      "post": {
        "tags": [
          "Spaces"
        ],
        "operationId": "Spaces_CreateSpace",
        "parameters": [
          {
            "name": "spaceName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          },
          {
            "name": "parentSpaceName",
            "in": "query",
            "required": false,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "The space was successfully created."
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure. The request failed. Space name is invalid.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      },
      "delete": {
        "tags": [
          "Spaces"
        ],
        "operationId": "Spaces_DeleteSpace",
        "parameters": [
          {
            "name": "spaceName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          },
          {
            "name": "noDelete",
            "in": "query",
            "required": false,
            "schema": {
              "type": "boolean",
              "default": false
            }
          },
          {
            "name": "wait",
            "in": "query",
            "required": false,
            "schema": {
              "type": "boolean",
              "default": false
            }
          }
        ],
        "responses": {
          "200": {
            "description": "The space was successfully deleted."
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure. The request failed. Specified space cannot be deleted.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Spaces/{namespaceName}": {
      "put": {
        "tags": [
          "Spaces"
        ],
        "operationId": "Spaces_CreateSpaceFromNamespace",
        "parameters": [
          {
            "name": "namespaceName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          },
          {
            "name": "parentSpaceName",
            "in": "query",
            "required": false,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "The space was successfully created"
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure. The request failed. Bad request.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Spaces": {
      "get": {
        "tags": [
          "Spaces"
        ],
        "operationId": "Spaces_ListSpaces",
        "responses": {
          "200": {
            "description": "The request was successful; spaces are returned.",
            "content": {
              "text/plain": {
                "schema": {
                  "uniqueItems": false,
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/Space"
                  }
                }
              },
              "application/json": {
                "schema": {
                  "uniqueItems": false,
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/Space"
                  }
                }
              },
              "text/json": {
                "schema": {
                  "uniqueItems": false,
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/Space"
                  }
                }
              }
            }
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure.",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Traces/request": {
      "post": {
        "tags": [
          "Traces"
        ],
        "operationId": "Traces_ReportRequestAsync",
        "responses": {
          "200": {
            "description": "The request was successful"
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/Traces/response": {
      "post": {
        "tags": [
          "Traces"
        ],
        "operationId": "Traces_ReportResponseAsync",
        "responses": {
          "200": {
            "description": "The request was successful"
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    },
    "/api/v3.2/UserPod": {
      "post": {
        "tags": [
          "UserPod"
        ],
        "operationId": "UserPod_CreatePodPatchOperation",
        "responses": {
          "200": {
            "description": "The request was successful; review response is returned.",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/UserPodReviewResponse"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/UserPodReviewResponse"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/UserPodReviewResponse"
                }
              }
            }
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure.",
            "content": {
              "text/plain": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              },
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              },
              "text/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        },
        "requestBody": {
          "required": false,
          "x-ms-requestBody-name": "request",
          "content": {
            "application/json-patch+json": {
              "schema": {
                "$ref": "#/components/schemas/UserPodReviewRequest"
              }
            },
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/UserPodReviewRequest"
              }
            },
            "text/json": {
              "schema": {
                "$ref": "#/components/schemas/UserPodReviewRequest"
              }
            },
            "application/*+json": {
              "schema": {
                "$ref": "#/components/schemas/UserPodReviewRequest"
              }
            }
          }
        },
        "x-ms-requestBody-index": 0
      }
    },
    "/api/v3.2/Workspace/spaces/{spaceName}/services/{serviceName}/workspaces/{name}": {
      "get": {
        "tags": [
          "Workspace"
        ],
        "operationId": "Workspace_DownloadSource",
        "parameters": [
          {
            "name": "spaceName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          },
          {
            "name": "serviceName",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          },
          {
            "name": "name",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "The request was successful. Code was downloaded.",
            "content": {
              "application/gzip": {
                "schema": {
                  "type": "file"
                }
              }
            }
          },
          "401": {
            "description": "The caller failed authentication"
          },
          "403": {
            "description": "The caller does not have proper access to this API"
          },
          "default": {
            "description": "Error response describing the reason for operation failure. The request failed. Requested code was not found.",
            "content": {
              "application/gzip": {
                "schema": {
                  "$ref": "#/components/schemas/ExecServiceErrorResponse"
                }
              }
            }
          }
        }
      }
    }
  },
  "tags": [],
  "components": {
    "schemas": {
      "PingResponse": {
        "type": "object",
        "properties": {
          "time": {
            "format": "date-time",
            "type": "string"
          },
          "connection": {
            "type": "string"
          },
          "buildVersion": {
            "type": "string"
          }
        }
      },
      "Space": {
        "required": [
          "name",
          "parentSpaceName"
        ],
        "type": "object",
        "properties": {
          "name": {
            "type": "string"
          },
          "parentSpaceName": {
            "type": "string"
          },
          "isDevSpacesReady": {
            "type": "boolean"
          }
        }
      },
      "UserPodReviewRequest": {
        "type": "object",
        "properties": {
          "namespace": {
            "type": "string"
          },
          "parentnamespace": {
            "type": "string"
          },
          "grandparentnamespace": {
            "type": "string"
          },
          "pod": {
            "type": "object"
          }
        }
      },
      "UserPodReviewResponse": {
        "type": "object",
        "properties": {
          "jsonpatches": {
            "uniqueItems": false,
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/JsonPatchOperation"
            }
          }
        }
      },
      "JsonPatchOperation": {
        "type": "object",
        "properties": {
          "op": {
            "type": "string"
          },
          "path": {
            "type": "string"
          },
          "value": {
            "type": "object"
          }
        }
      },
      "ExecServiceErrorResponse": {
        "description": "Error response indicates that the service is not able to process the incoming request. The reason is provided in the error message.",
        "properties": {
          "error": {
            "$ref": "#/components/schemas/ErrorDetails",
            "description": "The details of the error."
          }
        }
      },
      "ErrorDetails": {
        "properties": {
          "code": {
            "description": "Status code for the error.",
            "type": "string",
            "readOnly": true
          },
          "message": {
            "description": "Error message describing the error in detail.",
            "type": "string",
            "readOnly": true
          },
          "target": {
            "description": "The target of the particular error.",
            "type": "string",
            "readOnly": true
          }
        }
      }
    }
  }
}